packages/react/src/Timeline/Timeline.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {PropsWithChildren, useMemo, useCallback} from 'react'
+import clsx from 'clsx'
+import styles from './Timeline.module.css'
+import {BaseProps} from '../component-helpers'
+import {useAnimation} from '../animation'
+import {Text} from '../Text'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/timeline/colors-with-modes.css'
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/timeline/base.css'
+
+export type TimelineProps = BaseProps<HTMLUListElement>
+
+const _TimelineRoot = ({animate, className, children, ...rest}: PropsWithChildren<TimelineProps>) => {
+  const {classes: animationClasses, styles: animationInlineStyles} = useAnimation(animate)
+
+  const timelineClassName = clsx(animationClasses, styles.Timeline, className)
+
+  return (
+    <ul className={timelineClassName} style={{...animationInlineStyles}} {...rest}>
+      {children}
+    </ul>
+  )
+}
+
+export type TimelineItemProps = BaseProps<HTMLLIElement>
+
+const Item = ({className, children, ...rest}: PropsWithChildren<TimelineItemProps>) => {
+  const itemClassName = clsx(styles['Timeline__item'], className)
+  const childrenArray = useMemo(() => React.Children.toArray(children), [children])
+
+  const getConditionalVariant = useCallback(() => {
+    if (childrenArray.some(child => React.isValidElement(child) && child.type === 'em')) {
+      return 'muted'
+    }
+    return 'default'
+  }, [childrenArray])
+
+  const defaultColor = childrenArray.length === 1 ? 'default' : getConditionalVariant()
```

Comment:
I think you'll see a drop in unit test coverage with this new logic. To bring it back up, you could repurpose the tests I've added [here](https://github.com/primer/brand/pull/376/files#diff-1472b59f93a0204453d822a82224c9ebef8f1021d6a2e6058b41a5a1a312cac5R168-R193) for section intro. Should bring it back up to 100% 👍